openstack/networking/v2/ports/requests.go

```diff
@@ -130,7 +131,37 @@ type CreateOpts struct {
 
 // ToPortCreateMap builds a request body from CreateOpts.
 func (opts CreateOpts) ToPortCreateMap() (map[string]interface{}, error) {
-	return gophercloud.BuildRequestBody(opts, "port")
+	body, err := gophercloud.BuildRequestBody(opts, "port")
+	if err != nil {
+		return nil, err
+	}
+
+	return AddValueSpecs(body)
+}
+
+// AddValueSpecs expands the 'value_specs' object and removes 'value_specs'
+// from the request body. It will return error if the value specs would overwrite
+// an existing field or contains forbidden keys.
+func AddValueSpecs(body map[string]interface{}) (map[string]interface{}, error) {
+	// Banned the same as in heat. See https://github.com/openstack/heat/blob/dd7319e373b88812cb18897f742b5196a07227ea/heat/engine/resources/openstack/neutron/neutron.py#L59
```

Comment:
Would probably be better to link to opendev.org instead of GitHub mirrors, no big deal.

Reply:
Ah my bad :facepalm: 
Just say the word and I'll update...